### ISSUE TITLE
`ReferenceError: fetch is not defined` - calling unsplash.photos.getPhoto

### ISSUE THREAD

**Issue by alx-andru** [unsplash/unsplash-js#35]
Hi, 

I'm using unsplash in node.js and I'm coming up with following error message:

```
app[web.1]: ReferenceError: fetch is not defined
app[web.1]:     at Unsplash.request (/app/node_modules/unsplash-js/lib/unsplash.js:79:14)
app[web.1]:     at Object.getPhoto (/app/node_modules/unsplash-js/lib/methods/photos.js:77:20)

```
calling following code snippet:
```
unsplash.photos.getPhoto(imageId, 200, 1080, [0, 0, 1920, 1080])
    .then(toJson)
    .then(json => {
      console.log(json);
    });
```

#### Steps to Reproduce
In node.js imported and initialized unsplash-js

```
import Unsplash from 'unsplash-js';

const unsplash = new Unsplash({
  applicationId: "...",
  secret: "...",
  callbackUrl: "..."
});
```
determined a valid unsplash ID through my application logic and tried to call the API to retrieve additional info:

```
unsplash.photos.getPhoto(imageId, 200, 1080, [0, 0, 1920, 1080])
    .then(toJson)
    .then(json => {
      console.log(json);
    });
```

#### Observed Behaviour
Exception is thrown

> _Image or video please._
`N/A`

#### Technical Notes
Node.js hostend on heroku and started via 

`web: babel-node --presets es2015 main.js`

-

**Comment by joshwcomeau** [comment]
Hey there,

So, Unsplash-js depends on `fetch`, an upcoming JS API for making async requests.

We didn't want to bundle a polyfill as a dependency, because it isn't required when targeting modern browsers. 

It should work if you use it with [isomorphic-fetch](https://github.com/matthew-andrews/isomorphic-fetch). Require it before you require `unsplash-js`.

Let me know if this works for you!

**Comment by alx-andru** [comment]
Hi @joshwcomeau , 

thanks for the prompt reply,  I added

`npm install --save isomorphic-fetch es6-promise`

as suggested and required it at the top, worked! 

```
require('es6-promise').polyfill();
require('isomorphic-fetch');
```

**Comment by joshwcomeau** [comment]
Woohoo! Glad to hear it :)

**Comment by kosherpickle** [comment]
I am having trouble getting the solution to work. 

Here is my code: 
`require('es6-promise').polyfill();
require('isomorphic-fetch');
import Unsplash, { toJson } from 'unsplash-js';`

It works fine in Chrome (as it did before adding the requires), but does not work in Safari. I am getting an error "Failed to load resource: the server responded with a status of 404 (Not Found)" in regards to es6-promise.map. I have tried all sorts of variations on "import" rather than require, but to no avail (not even sure that this is the correct route to solving the issue). 

Before I added the requires, I was getting an error of "ReferenceError: Can't find variable: fetch" in Safari.

Obviously just learning; any help would be appreciated.

**Comment by joshwcomeau** [comment]
Hey @kosherpickle,

Sorry to hear that it's giving you trouble!

So, it _sounds_ like an issue related to `es6-promise`. I'm not sure how to debug that; can you share the project with me? If I can run it locally I can tinker with it.

For targeting Safari, though, you shouldn't need a promise polyfill; promises have been [supported for a while](http://caniuse.com/#feat=promises) in Safari. Does it work if you remove the es6-promise polyfill?

**Comment by kosherpickle** [comment]
I appreciate you taking the time to help me figure this out. I am new at quite a bit of this, so I imagine I have done something incorrectly. And I should also say that I am new to git as well. I just pushed a commit that I hope will allow you to clone the project; it is here: https://github.com/kosherpickle/first-tutorial
Please let me know if you need anything else. And I should say that it did not work without the es6-promise polyfill either.

**Comment by joshwcomeau** [comment]
Hey @kosherpickle,

No worries :)

So, I just checked and I see your `build` folder is checked in, but nothing else. Typically the `build` directory has the bundled files, but you should also have all of the source files. If you can commit all the other files related to this project, that'd be helpful.

**Comment by kosherpickle** [comment]
Ok. I think I have the other files committed now. Thanks for your patience.
And let me know if I still need to do something else.

**Comment by joshwcomeau** [comment]
hey @kosherpickle,

Ok, so I played around with your project.

I didn't get any errors as it is currently, in any browser. When I tried to search, however, it would reload the page.

This is because you have a `<button>` inside a `<form>`, and you aren't doing anything to prevent the default behaviour of submitting the form. By changing the `<form>` to a `<div>`, it works in Chrome and Safari:

```
function makeFormElement(wrapperID, formID, formClass) {
  let wrapper = document.getElementById(wrapperID);
  let form = document.createElement("div");
```

If you're still getting an error about the dependency in Safari, I would try deleting your `node_modules` directory and re-installing with `npm install`.
